test/serve/basic/serve-basic.test.js

```diff
@@ -371,18 +371,16 @@ describe("basic serve usage", () => {
   });
 
   it("should log error on using '--watch' flag with serve", async () => {
-    const { exitCode, stdout, stderr } = await runWatch(testPath, ["serve", "--watch"]);
+    const { stdout, stderr } = await runWatch(__dirname, ["serve", "--watch", "--port", port]);
 
-    expect(exitCode).toBe(0);
     expect(normalizeStderr(stderr)).toMatchSnapshot("stderr");
     expect(stdout).toContain("HotModuleReplacementPlugin");
     expect(stdout).toContain("main.js");
   });
 
   it("should log warning on using '-w' alias with serve", async () => {
-    const { exitCode, stdout, stderr } = await runWatch(testPath, ["serve", "-w"]);
+    const { stdout, stderr } = await runWatch(__dirname, ["serve", "-w", "--port", port]);
 
-    expect(exitCode).toBe(0);
```

Comment:
We don't test on the exit code for the `serve` because we exit forcefully and the exit code is different on Linux/windows.

https://github.com/webpack/webpack-cli/blob/eaca5d27e1b9cc75b0395738c700c84033630a5d/test/utils/test-utils.js#L31-L37